File: app/src/main/java/com/dailyserviceapp/data/Customer.java
package com.dailyserviceapp.data;

import com.google.firebase.Timestamp;
import com.google.firebase.firestore.DocumentId;

public class Customer {
    @DocumentId
    private String id;

    private String name;
    private String phone;
    private String address;
    private String serviceType;
    private double ratePerUnit;
    private Timestamp createdAt;

    public Customer() {
        // Firestore requires a public no-arg constructor
    }

    public Customer(String name, String phone, String address, String serviceType, double ratePerUnit, Timestamp createdAt) {
        this.name = name;
        this.phone = phone;
        this.address = address;
        this.serviceType = serviceType;
        this.ratePerUnit = ratePerUnit;
        this.createdAt = createdAt;
    }

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getPhone() {
        return phone;
    }

    public String getAddress() {
        return address;
    }

    public String getServiceType() {
        return serviceType;
    }

    public double getRatePerUnit() {
        return ratePerUnit;
    }

    public Timestamp getCreatedAt() {
        return createdAt;
    }

    public void setId(String id) {
        this.id = id;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public void setServiceType(String serviceType) {
        this.serviceType = serviceType;
    }

    public void setRatePerUnit(double ratePerUnit) {
        this.ratePerUnit = ratePerUnit;
    }

    public void setCreatedAt(Timestamp createdAt) {
        this.createdAt = createdAt;
    }
}

File: app/src/main/java/com/dailyserviceapp/data/models/Customer.kt
package com.dailyserviceapp.data.models

import com.google.firebase.Timestamp
import com.google.firebase.firestore.DocumentId

data class Customer(
    @DocumentId var id: String? = null,
    var name: String? = null,
    var phone: String? = null,
    var address: String? = null,
    var area: String? = null,
    var serviceType: String? = null,
    var ratePerUnit: Double = 0.0,
    var defaultQuantity: Double = 1.0,
    var lentAmount: Double = 0.0,
    var providerId: String? = null,
    var status: String? = "ACTIVE",
    var notes: String? = null,
    @get:JvmName("isOnVacation") @set:JvmName("setOnVacation") var onVacation: Boolean = false,
    var startDate: Timestamp? = null,
    var createdAt: Timestamp? = null
) {
    constructor(
        name: String?,
        phone: String?,
        address: String?,
        serviceType: String?,
        ratePerUnit: Double,
        createdAt: Timestamp?
    ) : this() {
        this.name = name
        this.phone = phone
        this.address = address
        this.serviceType = serviceType
        this.ratePerUnit = ratePerUnit
        this.createdAt = createdAt
        this.status = "ACTIVE"
        this.defaultQuantity = 1.0
        this.lentAmount = 0.0
    }
}
